src/test/java/com/puppycrawl/tools/checkstyle/internal/TestUtils.java

```diff
@@ -56,4 +66,46 @@ public static PackageObjectFactory getPackageObjectFactory() throws CheckstyleEx
         final Set<String> packageNames = PackageNamesLoader.getPackageNames(cl);
         return new PackageObjectFactory(packageNames, cl);
     }
+
+    /**
+     * Finds node of specified type among root children on any deep level.
+     * @param root      DetailAST
+     * @param predicate predicate
+     * @return {@link Optional} of {@link DetailAST} node which matches the predicate.
+     */
+    public static Optional<DetailAST> findTokenInASTByPredicate(DetailAST root,
+                                                                Predicate<DetailAST> predicate) {
+        DetailAST curNode = root;
+        while (!predicate.test(curNode)) {
+            DetailAST toVisit = curNode.getFirstChild();
+            while (curNode != null && toVisit == null) {
+                toVisit = curNode.getNextSibling();
+                if (toVisit == null) {
+                    curNode = curNode.getParent();
+                }
+            }
+
+            if (curNode == toVisit || curNode.equalsTree(root.getParent())) {
```

Comment:
You fixed if current node goes to parent, but what about if it goes to the next sibling of root?